changelog_unreleased/javascript/pr-8673.md

```diff
@@ -0,0 +1,81 @@
+#### Allow more arrow function print on same line with arrow ([#8673](https://github.com/prettier/prettier/pull/8673) by [@fisker](https://github.com/fisker))
+
+<!-- prettier-ignore -->
+```js
+// Input
+() => foo({
```

Comment:
What is the point of these examples where arrow functions are used like statements? Nobody writes code like this.

Reply:
On the other hand, this is still formatted the old way:

**Prettier pr-8673**
[Playground link](https://deploy-preview-8673--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEAzTBeTACgEp8A+AHSkxwgkOCppoGs4BPJTANwEMAbAK5wANE0wBfYgG4QIkBAAOMAJbpkoXgCctEAO4AFbQjTIQAvb3an5AIy28wbGAGVFjlVADmyGFuHycAC2tnAAJmHhADK83oK8XnAAYhBaQbwwqt5mvIIwEHIgABYwQfwA6kUq8GjuYHAuJtUq3NXsZmBoNiCeaHBaMAYOXunI2AJ98gBWaAAeAEIOTnCuvEFwUZ5wYxOiIDOzLp5e-HAAioIQ8Dv8kyDuWn1aZra8ofyFilqeMOUqYTAisgABwABnkXwgfXKDkUZi+cCe3G28gAjpd4EMlKYULw0ABaKBwcLhQpaODolTkoYJUZIca3PZ9IIqXz+JnHU4XK7bem7eQwN5-AFApAAJgFDhU-GOAGEIEE6SBEQBWQqCPoAFTeOIZd24wgAklBIrAXGBvsoAIImlwwdinG59CQSIA)
```sh
--parser babel
```

**Input:**
```jsx
const f = () =>
  foo({
    key: value,
  });
```

**Output:**
```jsx
const f = () =>
  foo({
    key: value,
  });

```

Overall, this change looks definitely good for arrow functions without parameters. I'd say arrow functions without parameters should never get a line break after the arrow. But I'm not sure about arrow functions with parameters, especially with multiple parameters.